lazy_theta_star_p_planner/include/lazy_theta_star_p_planner/lazy_theta_star_p.hpp

```diff
@@ -0,0 +1,238 @@
+   * @param raw_points used to return the path  thus found
+   * @param curr_id sends in the index of the goal coordinate, as stored in nodes_position
+   */
+  void backtrace(std::vector<coordsW> & raw_points, int curr_id);
+
+  /**
+   * @brief performs the line of sight check using Bresenham's Algorithm,
+   *            and has been modified to calculate the traversal cost incurred in a straight line path between
+   *            the two points whose coordinates are (x0, y0) and (x1, y1)
+   * @param sl_cost is used to return the cost thus incurred
+   * @return true if a line of sight exists between the points
+   */
+  bool losCheck(
+    const int & x0, const int & y0, const int & x1, const int & y1,
+    double & sl_cost);
+
+  void initializePosn(int size_inc = 0);
+
+  void setContainers();
+
+  double dist(const int & ax, const int & ay, const int & bx, const int & by)
+  {
+    return std::hypot(ax - bx, ay - by);
+  }
+
+  double getEucledianCost(const int & ax, const int & ay, const int & bx, const int & by)
+  {
+    return euc_tolerance_ * dist(ax, ay, bx, by);
+  }
+
+  double getCellCost(const int & cx, const int & cy) const
+  {
+    return costmap_->getCost(cx + 1, cy + 1);
+  }
+
+  /**
+   * @brief for the point(cx, cy) its traversal cost is calculated by <parameter>*(<actual_traversal_cost_from_costmap>)^2/(<max_cost>)^2
+   * @return the traversal cost thus calculated
+   */
+  double getTraversalCost(const int & cx, const int & cy)
+  {
+    double curr_cost = getCellCost(cx, cy);
+    return costmap_tolerance_ * curr_cost * curr_cost / LETHAL_COST / LETHAL_COST;
+  }
+  /**
+   * @brief it is an overloaded function to ease in cost calculations while performing the LOS check
+   * @param cx
+   * @param cy
+   * @param cost denotes the total straight line traversal cost, adds the traversal cost for the node (cx, cy) at every instance
+   */
+  bool isSafe(const int & cx, const int & cy, double & cost) const
+  {
+    double curr_cost = getCellCost(cx, cy);
+    if (curr_cost < LETHAL_COST) {
+      cost += costmap_tolerance_ * curr_cost * curr_cost / LETHAL_COST / LETHAL_COST;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  void addIndex(const int & cx, const int & cy, const int & id_this)
+  {
+    node_position[size_x_ * cy + cx] = id_this;
+  }
+
+  void getIndex(const int & cx, const int & cy, int & id_this)
+  {
+    id_this = node_position[size_x_ * cy + cx];
+  }
+
+  bool withinLimits(const int & cx, const int & cy) const
+  {
+    return cx >= 0 && cx < size_x_ && cy >= 0 && cy < size_y_;
+  }
+
+  bool isGoal(const int & cx, const int & cy) const
+  {
+    return cx == dst.x && cy == dst.y;
+  }
+
+  void addToNodesData(const int & id_this)
+  {
+    if (nodes_data.size() <= static_cast<unsigned int>(id_this)) {
+      nodes_data.push_back({});
+    } else {
+      nodes_data[id_this] = {};
+    }
+  }
+
+  void clearQueue()
```

Comment:
https://github.com/SteveMacenski/navigation2/blob/nav2_smac_planner/smac_planner/src/a_star.cpp#L545 this ends up being a tad bit faster

Reply:
That isn't intuitive, I learned that from testing while optimizing the codebase